Paintroid/src/main/java/org/catrobat/paintroid/presenter/LayerPresenter.kt

```diff
@@ -167,21 +167,23 @@ class LayerPresenter(
     }
 
     override fun hideLayer(position: Int) {
-        val destinationLayer = getDestinationLayer(position, false)
-        drawingSurface?.refreshDrawingSurface()
-        if (model.currentLayer == destinationLayer) {
-            defaultToolController?.switchTool(ToolType.HAND, false)
-            bottomNavigationViewHolder?.showCurrentTool(ToolType.HAND)
+        getDestinationLayer(position, false)?.let { layer ->
+            drawingSurface?.refreshDrawingSurface()
+            if (model.currentLayer == layer) {
+                defaultToolController?.switchTool(ToolType.HAND, false)
+                bottomNavigationViewHolder?.showCurrentTool(ToolType.HAND)
+            }
         }
     }
 
     override fun unhideLayer(position: Int, viewHolder: LayerContracts.LayerViewHolder) {
-        val destinationLayer = getDestinationLayer(position, true)
-        viewHolder.updateImageView(destinationLayer.bitmap, true)
-        drawingSurface?.refreshDrawingSurface()
-        if (model.currentLayer == destinationLayer) {
-            defaultToolController?.switchTool(ToolType.BRUSH, false)
-            bottomNavigationViewHolder?.showCurrentTool(ToolType.BRUSH)
+        getDestinationLayer(position, true)?.let { layer ->
+            viewHolder.updateImageView(layer.bitmap, true)
+            drawingSurface?.refreshDrawingSurface()
```

Comment:
check suggestion at line 170

Reply:
will also refer to the discussion at line 170